From 4b619600b67c7c83c123c2756db8be6b9fe7d3cf Mon Sep 17 00:00:00 2001
From: Alex Gonzalez <alex.gonzalez@digi.com>
Date: Mon, 17 Dec 2018 16:40:30 +0100
Subject: [PATCH] ARM: dts: ccimx6ulstarter: Add support for the Focaltech
 FT6236 cap touch

This is used in some of Adafruit's displays.

Signed-off-by: Alex Gonzalez <alex.gonzalez@digi.com>
---
 arch/arm/boot/dts/imx6ul-ccimx6ulstarter-wb.dts |  4 ++++
 arch/arm/boot/dts/imx6ul-ccimx6ulstarter.dtsi   | 17 +++++++++++++++++
 2 files changed, 21 insertions(+)

diff --git a/arch/arm/boot/dts/imx6ul-ccimx6ulstarter-wb.dts b/arch/arm/boot/dts/imx6ul-ccimx6ulstarter-wb.dts
index 345e0274020e..be8c93db5738 100644
--- a/arch/arm/boot/dts/imx6ul-ccimx6ulstarter-wb.dts
+++ b/arch/arm/boot/dts/imx6ul-ccimx6ulstarter-wb.dts
@@ -50,6 +50,10 @@
 	status = "okay";
 };
 
+&ft6236 {
+        status = "okay";
+};
+
 /* CAN1 (Raspberry PI Expansion header) */
 &can1 {
 	status = "disabled";
diff --git a/arch/arm/boot/dts/imx6ul-ccimx6ulstarter.dtsi b/arch/arm/boot/dts/imx6ul-ccimx6ulstarter.dtsi
index e191c94d3cee..a8cab6c733f0 100644
--- a/arch/arm/boot/dts/imx6ul-ccimx6ulstarter.dtsi
+++ b/arch/arm/boot/dts/imx6ul-ccimx6ulstarter.dtsi
@@ -61,6 +61,17 @@
 	scl-gpios = <&gpio1 0 GPIO_ACTIVE_HIGH>;
 	sda-gpios = <&gpio1 1 GPIO_ACTIVE_HIGH>;
 	status = "disabled";
+
+	ft6236: ft6236@38 {
+		compatible = "focaltech,ft6236";
+		reg = <0x38>;
+		pinctrl-0 = <&pinctrl_focaltech_touch>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <12 IRQ_TYPE_EDGE_RISING>;
+		touchscreen-size-x = <240>;
+		touchscreen-size-y = <320>;
+		status = "disabled";
+	};
 };
 
 &pwm1 {
@@ -188,6 +199,12 @@
 			>;
 		};
 
+		pinctrl_focaltech_touch: ft6236grp {
+			fsl,pins = <
+				MX6UL_PAD_LCD_DATA07__GPIO3_IO12       0xf030
+			>;
+		};
+
 		pinctrl_i2c2: i2c2grp {
 			fsl,pins = <
 				MX6UL_PAD_GPIO1_IO00__I2C2_SCL 0x4001b8b0
